cypress_test/integration_tests/mobile/calc/undo_redo_spec.js

```diff
@@ -0,0 +1,81 @@
+/* global describe it cy beforeEach require afterEach */
+
+var helper = require('../../common/helper');
+var mobileHelper = require('../../common/mobile_helper');
+var calcHelper = require('../../common/calc_helper');
+
+describe('Editing Operations', function() {
+	var testFileName = 'undo_redo.ods';
+
+	beforeEach(function() {
+		helper.beforeAll(testFileName, 'calc');
+
+		// Click on edit button
+		mobileHelper.enableEditingMobile();
+	});
+
+	afterEach(function() {
+		helper.afterAll(testFileName, this.currentTest.state);
+	});
+
+	function undo() {
+		calcHelper.dblClickOnFirstCell();
+
+		helper.typeIntoDocument('Hello World');
+
+		//if we don't wait tests in CLI is failing
+		cy.wait(500);
+
+		cy.get('#tb_actionbar_item_undo').click();
+
+		helper.selectAllText();
+
+		helper.expectTextForClipboard('Hello Worl');
+	}
+
+	it('Undo', function() {
+		undo();
+	});
+
+	it('Redo', function() {
+		undo();
+
+		cy.get('#tb_actionbar_item_redo').click();
+
+		helper.selectAllText();
+
+		helper.expectTextForClipboard('Hello World');
+	});
+
+	it('Repair Document', function() {
+		helper.typeIntoDocument('Hello World');
+
+		helper.typeIntoDocument('{enter}');
+
+		calcHelper.dblClickOnFirstCell();
+
+		helper.clearAllText();
+
+		helper.typeIntoDocument('Hello');
+
+		helper.typeIntoDocument('{enter}');
+
+		cy.get('#toolbar-hamburger')
```

Comment:
would be good to have something text based(menu name) rather than counting nodes, just in case if anything is added in menu this would break. Don't need to spend much time on it if you don't find a solution then this would work.

Reply:
I already tried many things before doing this approach nothing was working for me